doc/OCP_LOCK.md

```diff
@@ -0,0 +1,256 @@
+
+OCP L.O.C.K. is removing the need to destroy storage devices and providing the security to address these issues by:
+
+- Preventing leakage of media keys via firmware vulnerabilities or side channels;
+- Binding media keys to a set of securely-provisioned access keys; and
+- Enabling attestable “hard” purge via erasure of fuse secrets.
+
+## Overview
+
+Self-encrypting drives (SEDs) store data encrypted to media encryption keys (MEKs). SEDs include the following building blocks:
+
+- The storage media that holds data at rest.
+- A hardware cryptographic engine that performs line-rate encryption and decryption of data as it enters and exits the drive.
+- A controller that manages the lifecycle of MEKs.
+
+MEKs may be bound to user credentials, which the host must provide to the drive in order for the associated data to be readable. A given MEK may be bound to one or more credentials. This model is captured in the TCG Opal [specification](https://trustedcomputinggroup.org/resource/storage-work-group-storage-security-subsystem-class-opal/).
+
+MEKs, or other keys from which MEKs are derived, may be injected into the drive and tagged with address range metadata, such that subsequent I/Os which target that address range will be encrypted to that injected MEK. This model is captured in the TCG Key Per I/O (KPIO) [specification](https://trustedcomputinggroup.org/resource/tcg-storage-security-subsystem-class-ssc-key-per-i-o/).
+
+MEKs may be securely erased, to effectively erase all data which was encrypted to the MEK. To erase an MEK, it is sufficient for the controller to erase the MEK itself or a key from which it was derived.
+
+In an SED that takes Caliptra with L.O.C.K. features enabled, Caliptra will act as a Key Management Block (KMB). The KMB will be the only entity that can read MEKs and program them into the SED's cryptographic engine. The KMB will expose services to controller firmware which will allow the controller to transparently manage each MEK's lifecycle, without being able to access the raw MEK itself.
+
+## Threat model
+
+The protected asset is the user data stored at rest on the drive. The adversary profile extends up to nation-states in terms of capabilities.
+
+Adversary capabilities include:
+
+- Interception of a storage device in the supply chain.
+- Theft of a storage device from a data center.
+- Destructively inspecting a stolen device.
+- Running arbitrary firmware on a stolen device.
```

Comment:
Gamil had an earlier comment about adding insider attack in scope with ratchet

Reply:
Added a note. cc @ericeilertson for review.